test/vm-download

```diff
@@ -18,78 +18,79 @@
 # You should have received a copy of the GNU Lesser General Public License
 # along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
 
-SELF=vm-download
-. ./testlib.sh
-
-URL=https://fedorapeople.org/groups/cockpit/testdata/images
-
-TEST_DATA=${TEST_DATA:-$PWD}
-
-set -euf
-
-usage()
-{
-    echo >&2 "usage: $SELF [-f FLAVOR]"
-}
-
-flavor=cockpit
-
-case ${1:-} in
--f|--flavor)
-	if [ $# -lt 2 ]; then
-		usage
-		exit 2
-	fi
-	flavor=$2
-	shift 2
-	;;
---flavor=*)
-	flavor=${1#--machine=}
-	shift
-	;;
---help|-h)
-	usage
-	exit 0
-	;;
-esac
-
-os=$(python -c "import json; print json.load(open('guest/$flavor.conf', 'r'))['os']" 2>/dev/null || true)
-os=${os:-$TEST_OS}
-
-tag=$(python -c "import json; print json.load(open('guest/$flavor.conf', 'r'))['tags']['$os']" 2>/dev/null || true)
-tag=${tag:-0}
-
-TEST_IMAGE=$flavor-$os-$TEST_ARCH-$tag
-
-checksum_file=$TEST_IMAGE-checksum
-
-if [ ! -e "$TEST_DATA/images" ]; then
-    mkdir "$TEST_DATA/images"
-fi
-
-cd "$TEST_DATA/images"
-
-if ! curl -f -s "$URL/$checksum_file" -o "$checksum_file.remote"; then
-    if [ ! -f "$checksum_file" ]; then
-        echo Downloading $URL/$checksum_file failed.
-        exit 1
-    else
-        echo Downloading $URL/$checksum_file failed, using existing image.
-        exit 0
-    fi
-fi
-
-if [ -f "$checksum_file" ] && cmp "$checksum_file.remote" "$checksum_file"; then
-    rm -f "$checksum_file.remote"
-    exit 0
-fi
-
-cat "$checksum_file.remote" | while read sum file; do
-  if [ ! -f "$file" ] || ! echo $sum "$file" | sha256sum --status --check; then
-      echo Downloading $file
-      curl "$URL/$file.xz" | xz -d >"$file.partial" && mv "$file.partial" "$file"
-  fi
-done
-
-sha256sum --strict --quiet --check "$checksum_file.remote"
-mv "$checksum_file.remote" "$checksum_file"
+DOWNLOAD = "https://fedorapeople.org/groups/cockpit/images/"
+
+import argparse
+import os
+import subprocess
+import shutil
+import sys
+import tempfile
+
+BASE = os.path.dirname(__file__)
+IMAGES = os.path.join(BASE, "images")
```

Comment:
why change this from using TEST_DATA? https://github.com/cockpit-project/cockpit/pull/3037/files#diff-82bca57c3ec28019b40b2e8fcd72cb92L26